docs/proposals/emptydir-pv.md

```diff
@@ -0,0 +1,151 @@
+* Every pod referencing this PV must land on the same node (this concept has been referred to as "data gravity").
+* Once this PV is bound to a node, it cannot get unbound till a human intervenes or the node is deleted
+* The PV must require low user interaction. It should be provisionable dynamically.
+* The directory backing the PV must get deleted when the PV is deleted.
+
+## Non goals
+* A local storage abstraction that survives extended node outages
+* Snapshotting or simply copying the local volume to another node
+
+## A new PV storage source
+
+As described above, the storage source for the PV must be a local volume. While we already allow hostPath as a PersistentVolumeSource, it is insufficient because:
+* user needs to specify the exact path, thus requires knowledge of the filesystem
+* hard to dynamically provision a hostpath
+* hostpath volumes are not cleaned up
+
+The `emptydir` abstraction solves these problems:
+* it is allocated in `/var/lib/kubelet/resource/uid/`, a location that is predictable but unknow till `resource` creation time
+* The node takes care of allocation, a dynamic provisioner just needs to create the PV
+* it is cleaned up when the pod dies (or in this case when the PV is deleted)
+
+We can create a new [PersistetVolumeSource](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L240) for EmptyDir. Users will request storage by creating PersistetVolumeClaims. If the PVC isn't bound to a PV, one will be created dynamically. The user can't control which node the storage for this PV comes from, or where in the node's filesystem it is allocated. Users can also hand create PVs bound to specific nodes (TODO: elaborate on QoS and what happens if there's no space on the node).
+
+## Scheduling constraints for data gravity
+
+The scheduler will schedule the pod paying attention to any referenced PVs. If an empty dir PV attached to an existing node is found, that trumps all scheduler predicates. Otherwise the scheduler makes a scheduling decision taking into account pod cpu/memory requests (and in the future, requested storage), attaches it to a node, and all subsequent scheduling decisions for any pod referencing the same PV evaluate to the same node. The scheduler "Attaches" a pv to a node by writing an annotation (TBD what actually writes this annotation, scheduler or kubelet, scheduler will at least bind the pod to the node). All pods referencing the PV that enter the "Running" state on the give node will be able to write to the volume. If the node enters the `NotReady` state, all pods referencing the PV attached to the node will remain in `Pending` till such time when an admin either removes the annotation, or rewrites it to point to another node. The directory allocated to the emptydir volume will be exported through the status field of the PV, so admins can yank the disk from a NotReady node and rsync data to the same directory on another node before rewriting the annotation.
```

Comment:
> 5) I didn't understand If the node enters the NotReady state, all pods referencing the PV attached to the node will remain in Pending till such time when an admin either removes the annotation, or rewrites it to point to another node. Since you say "remain in Pending" I assume you are talking about pods that are not yet running on the node? Why would you not allow them to schedule onto the node once the node is no longer NotReady? And are there any special rules about pods that are already running on the node (i.e. are they subject to the normal 5 minute eviction rule, or something else overrides that?)?

Wording wasn't clear. I meant both the cases you asked about, i.e: 
1. Pod-1 is Running attached to node1
2. Node1 is NotReady
3. 5 minutes later, pod-1 is deleted 
4. 10s later RC/petset controller recreates pod-1
5. pod-1 remains in pending because it's referencing a pv attached to node-1

In both cases (the above and the case where the scheduler dumps some pods on a node just as it enters NotReady) the pods will enter running if the node re-enters Ready. There are no special rules to avoid the 5m timeout, because the PV isn't getting rebound anyway.